src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/FormLoginFactory.php

```diff
@@ -22,14 +22,15 @@
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Johannes M. Schmitt <schmittjoh@gmail.com>
  */
-class FormLoginFactory extends AbstractFactory
+class FormLoginFactory extends AbstractFactory implements AuthenticatorFactoryInterface, EntrypointFactoryInterface
 {
     public function __construct()
     {
         $this->addOption('username_parameter', '_username');
         $this->addOption('password_parameter', '_password');
         $this->addOption('csrf_parameter', '_csrf_token');
         $this->addOption('csrf_token_id', 'authenticate');
+        $this->addOption('enable_csrf', false);
```

Comment:
In this new system, we have this new option `enable_csrf`. But in the old system, you enabled this by adding a `csrf_token_generator` under `form_login`. This leaves us in an odd spot: each option is *allowed* when you're using the other system, but does nothing (e.g. I can have a `csrf_token_generator` option under `form_login` while using the new system, but it does nothing).

Ideally - though I'm not sure how - those options would be conditionally added based on which system you're using.

Reply:
Yeah, I think we should add some `InvalidConfigurationException` in `create()` and `createAuthenticator()` when either of them is overwritten/configured, while not supported by that system.